openedx/core/djangoapps/xblock/rest_api/views.py

```diff
@@ -168,3 +171,86 @@ def cors_allow_xblock_handler(sender, request, **kwargs):  # lint-amnesty, pylin
 
 
 check_request_enabled.connect(cors_allow_xblock_handler)
+
+
+@view_auth_classes()
+class BlockFieldsView(APIView):
+    """
+    View to get/edit the field values of an XBlock as JSON (in the v2 runtime)
+
+    This class mimics the functionality of xblock_handler in block.py (for v1 xblocks), but for v2 xblocks.
+    However, it only implements the exact subset of functionality needed to support the v2 editors (from
+    the frontend-lib-content-components project). As such, it only supports GET and POST, and only the
+    POSTing of data/metadata fields.
+    """
+
+    @atomic
+    def get(self, request, usage_key_str):
+        """
+        retrieves the xblock, returning display_name, data, and metadata
+        """
+        try:
+            usage_key = UsageKey.from_string(usage_key_str)
+        except InvalidKeyError as e:
+            raise NotFound(invalid_not_found_fmt.format(usage_key=usage_key_str)) from e
+
+        block = load_block(usage_key, request.user)
+        block_dict = {
+            "display_name": get_block_display_name(block),
```

Comment:
maybe here is the place to note that you are separating out the display name from the metadata, because this is the aberration from the norm (one would expect that "title" would be metadata). Then you can get rid of the confusing messaging in other places.

Reply:
I suppose.
Note: this aberration is copied from the v1 version, and I assume it existed for some "good" reason, though one that's now undocumented.
From reading various bits of code, I think that it might not always be present in metadata (instead, only encoded in the OLX).

Reply:
added a comment.